python/paddle/fluid/compiler.py

```diff
@@ -335,13 +357,28 @@ def _compile(self, scope, place):
 
         self._scope = scope
         self._place = place
-        if self._is_data_parallel:
-            self._executor = self._compile_data_parallel(
-                use_cuda=isinstance(self._place, core.CUDAPlace),
-                scope=self._scope)
-        elif self._is_inference:
+
+        if self._is_inference:
             self._executor = self._compile_inference()
         else:
-            p = _place_obj(self._place)
-            self._executor = core.Executor(p)
+            if self._is_data_parallel:
+                self._places = self._get_places(self._place, self._places)
+            else:
+                self._places = [self._place]
```

Comment:
in this branch, `_is_data_parallel` is still `False`, then compiled, return to `program._compile()` in executor.py, is it get into the else branch? it will invoke `_run_program` and execute program by `Executor`, the `_compile_data_parallel` here is meaningless?

Reply:
Please read this introduction first: https://github.com/PaddlePaddle/Paddle/pull/18919#issue-302769160